man/jjcsformat_xx.Rd

```diff
@@ -2,19 +2,105 @@
 % Please edit documentation in R/jjcsformats.R
 \name{jjcsformat_xx}
 \alias{jjcsformat_xx}
-\title{Formatting of values}
+\alias{jjcsformat_pval_fct}
+\alias{jjcsformat_range_fct}
+\title{Utility for specifying custom formats}
 \usage{
-jjcsformat_xx(str, na_str = na_str_dflt)
+jjcsformat_xx(
+  str,
+  na_str = na_str_dflt,
+  na_str_dflt = "NE",
+  replace_na_dflt = TRUE
+)
+
+jjcsformat_pval_fct(alpha = 0.05)
+
+jjcsformat_range_fct(str)
 }
 \arguments{
-\item{str}{(\code{character})\cr the formatting that is required specified as a text string, eg "xx.xx"}
+\item{str}{(\code{string})\cr the format specifying the number of digits to be used,
+for the range values, e.g. \code{"xx.xx"}.}
+
+\item{na_str}{String for NA values.}
+
+\item{na_str_dflt}{Character to represent NA value}
 
-\item{na_str}{(\code{character})\cr Na string that will be passed from \code{formatters} into
-our formatting functions.}
+\item{replace_na_dflt}{logical(1). Should an \code{na_string} of "NA" within
+the formatters framework be overridden by \code{na_str_default}? Defaults to
+\code{TRUE}, as a way to have a different default na string behavior from the
+base \code{formatters} framework.}
+
+\item{alpha}{(\code{numeric})\cr the significance level to account for during rounding.}
 }
 \value{
-a formatting function with \code{"sas"}-style rounding.
+Either a supported format string, or a formatting function that can be
+used as format in \code{formatters::format_value}
+
+The p-value in the standard format. If \code{count} is 0, the format is \code{0}.
+If it is smaller than 0.001, then \verb{<0.001}, if it is larger than 0.999, then
+\verb{>0.999} is returned. Otherwise, 3 digits are used. In the special case that
+rounding from below would make the string equal to the specified \code{alpha},
+then a higher number of digits is used to be able to still see the difference.
+For example, 0.0048 is not rounded to 0.005 but stays at 0.0048 if \code{alpha = 0.005}
+is set.
+
+A function that formats a numeric vector with 4 elements:
+\itemize{
+\item minimum
+\item maximum
+\item censored minimum? (1 if censored, 0 if event)
+\item censored maximum? (1 if censored, 0 if event)
+The range along with the censoring information is returned as a string
+with the specified numeric format as \verb{(min, max)}, and the \code{+} is appended
```

Comment:
Should we have flexibility for users to specify the censoring identifier character with a default of "+" for J&J standards?

Reply:
@iaugusty , what do users do then if a different character is needed for identification of censored min/max (e.g. * instead of +)? Do they need to define a new function similar to jjcsformat_range_fct?